### PR TITLE
chore: repackage Document into component and viewer + tweaks

### PR DESCRIPTION
**Pull request by mwdchang**
### Summary
Giving the same treatment to Dataset as we had done previously for Models and Documents
- Repackage Dataset into a component and a viewer
- Add in tabs management system to the viewer
- Leverage the Document component across viewer and data-explorer
- First round of UI tweaks as per design spec: accordion mechanism + scrolling fixes

Note the scroll behaviour is a weird, will tidy it up after merging with the sliding-pane changes.

image

### Testing
- Should be able to preview dataset detail when clicking on dataset search result in data explorer
- Should be able to open multiple dataset tabs within the project dataset space

### CODE REVIEW

**Review by YohannParis — APPROVED** [review]
LGTM

File: packages/client/hmi-client/src/components/dataset/Dataset.vue [hunk 1]
```diff
@@ -45,42 +43,51 @@ const rawColumnNames = computed(() => {
...
+	return features;
+	// return features.map((f) => (f.display_name !== '' ? f.display_name : f.name));
 };
 </script>
 
 <template>
 	<section class="dataset">
 		<template v-if="dataset !== null">
-			<h3>{{ dataset?.name ?? '' }}</h3>
+			<div class="title">{{ dataset?.name ?? '' }}</div>
```
[the remark this answers is not in the document]
Comment (mwdchang) [reply packages/client/hmi-client/src/components/dataset/Dataset.vue:54]: I am following what had been done in other components, namely Document. 

I have also never find `h1-5` to be particularly meaningful as people redefine them as they see fit, e.g. an H2 is the equivalent of an H5 across sibling component. I think if you want to have semantic header-elements they need to be defined globally.

File: packages/client/hmi-client/src/components/dataset/Dataset.vue [hunk 2]
```diff
@@ -45,42 +43,51 @@ const rawColumnNames = computed(() => {
 const formatFeatures = (d: Dataset) => {
 	const features = d.annotations.annotations.feature ?? [];
 	if (!features || features.length === 0) return [];
-	return features.map((f) => (f.display_name !== '' ? f.display_name : f.name));
+	return features;
+	// return features.map((f) => (f.display_name !== '' ? f.display_name : f.name));
```
[the remark this answers is not in the document]
Comment (mwdchang) [reply packages/client/hmi-client/src/components/dataset/Dataset.vue:47]: Yep! I don't don't this space so was mucking around trying to decipher the data structures and things. Will clean it up.

File: packages/client/hmi-client/src/components/dataset/Dataset.vue (hunk 1 above)
[the remark this answers is not in the document]
Comment (mwdchang) [reply packages/client/hmi-client/src/components/dataset/Dataset.vue:52]: Will add it to the reno/debt list

**Review by YohannParis — APPROVED** [review]

**Pull request merged**